marimo/_ast/visitor.py

```diff
@@ -189,7 +194,11 @@ def refs(self) -> set[Name]:
     @property
     def deleted_refs(self) -> set[Name]:
         """Referenced names that were deleted with `del`."""
-        return set(name for name in self._refs if self._refs[name].deleted)
+        return set(
+            name
+            for name in self._refs
+            if any(ref.deleted for ref in self._refs[name])
```

Comment:
Hmm, I think maybe only the top block?

```python
x = 0
def fn():
    if False:
        z = x
        del x
x
```

Reply:
I think only possible with global keyword

Reply:
I think looking at all the blocks is right, because the `_refs` data structure keeps one `RefData` per name per block.

In your example, `x` is not added as a ref, so there's no issue.

I did however find a related bug, and just pushed up a fix and a couple of unit tests.